File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example3/AuctionParticipant.cs
namespace MediatorDesignPattern.Example3;

/// <summary>
/// Concrete Colleague - Regular auction participant
/// </summary>
public class AuctionParticipant : Bidder
{
    public AuctionParticipant(IAuctionMediator mediator, string name) 
        : base(mediator, name)
    {
    }

    public override void PlaceBid(decimal amount)
    {
        Console.WriteLine($"[{Name}] Placing bid: ${amount}");
        Mediator.PlaceBid(this, amount);
    }

    public override void ReceiveNotification(string message)
    {
        Console.WriteLine($"[{Name}] Notification: {message}");
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example3/src/AuctionParticipant.ts
import { Bidder } from './Bidder';
import { IAuctionMediator } from './IAuctionMediator';

export class AuctionParticipant extends Bidder {
  constructor(mediator: IAuctionMediator, name: string) {
    super(mediator, name);
  }

  placeBid(amount: number): void {
    console.log(`[${this.name}] Placing bid: $${amount}`);
    this.mediator.placeBid(this, amount);
  }

  receiveNotification(message: string): void {
    console.log(`[${this.name}] Notification: ${message}`);
  }
}
